File: 2104-sum-of-subarray-ranges/2104-sum-of-subarray-ranges.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var subArrayRanges = function(nums) {
  const n = nums.length;
  let sum = 0;
  
  for (let i = 0; i < n; i++) {
    let [min, max] = [Infinity, -Infinity];
    for (let j = i; j < n; j++) {
      min = Math.min(min, nums[j]);
      max = Math.max(max, nums[j]);
      sum += max - min;
    }
  }
  
  return sum;
};

File: 2104-sum-of-subarray-ranges/2104-sum-of-subarray-ranges.py
class Solution:
    def subArrayRanges(self, nums: List[int]) -> int:
        n, s = len(nums), 0
        for i in range(n):
            low, high = nums[i], nums[i]
            for j in range(i, n):
                low = min(low, nums[j])
                high = max(high, nums[j])
                s += high - low
        return s
      